doc/src/manual/types.md

```diff
@@ -1,73 +1,59 @@
 # [Types](@id man-types)
 
-Type systems have traditionally fallen into two quite different camps: static type systems, where
-every program expression must have a type computable before the execution of the program, and
-dynamic type systems, where nothing is known about types until run time, when the actual values
-manipulated by the program are available. Object orientation allows some flexibility in statically
-typed languages by letting code be written without the precise types of values being known at
-compile time. The ability to write code that can operate on different types is called polymorphism.
-All code in classic dynamically typed languages is polymorphic: only by explicitly checking types,
-or when objects fail to support operations at run-time, are the types of any values ever restricted.
-
-Julia's type system is dynamic, but gains some of the advantages of static type systems by making
-it possible to indicate that certain values are of specific types. This can be of great assistance
+Types in Julia establish distinctions between different kinds of data,
+and are used by the compiler to infer the intended use of those data.
+Programming languages have traditionally employed one of two quite different type systems:
+static type systems, where expressions must have a computable type before the execution of the program,
+and dynamic type systems, where types are only computed at run time,
+when the actual values manipulated by the program are available.
+Statically typed languages typically offer faster execution,
+at the cost of type annotations which must be explicitly added by the programmer.
+Dynamic typing, on the other hand, allows for polymorphism,
+which is the ability to write code that can operate on different types.
+Only by explicitly checking types, or when objects fail to support operations at run time,
+are the types of any values ever restricted.
+
+Julia's type system is dynamic, but gains some of the advantages of static type systems
+by allowing optional type annotations. This can be of great assistance
```

Comment:
Maybe:

> Type annotations are important for defining efficient new data structures, can help enforce correctness and clarity, and can sometimes improve efficiency (although in many contexts the Julia compiler infers types automatically). However, type annotations also play a central role in Julia because they control method dispatch: functions can be extended to new behaviors for different argument types.